packages/react/src/components/NumberInput/NumberInput.js

```diff
@@ -214,6 +246,23 @@ class NumberInput extends Component {
     this._inputRef = ref;
   };
 
+  /**
+   * @returns {boolean} `true` if the `value` prop is controlled.
+   */
+  isControlled() {
+    const { _lastIsControlled: lastIsControlled } = this;
+    const result = this.props.value !== undefined;
+    if (lastIsControlled !== undefined && lastIsControlled !== result) {
+      warning(
+        'A component is changing an uncontrolled `NumberInput` to be controlled. ' +
```

Comment:
The fact that we're trying to emulate 2 react errors feels like a code smell to me. How many errors, edge cases are we missing? This feels like a strong indication we're going the wrong direction with how we're attempting to manage the state of this component. We'd need to replicate this error in each component using this pattern for each piece of managed state?